src/hooks/useLoadDisciplines.ts

```diff
@@ -28,7 +32,12 @@ export const loadDisciplines = async (loadingInfo: RequestParams): Promise<Disci
   const url = getEndpoint(loadingInfo)
   const apiKey = isTypeLoadProtected(loadingInfo) ? loadingInfo.apiKey : loadingInfo.parent?.apiKey
   const response = await getFromEndpoint<ServerResponseDiscipline[]>(url, apiKey)
-  return response.map(item => formatDiscipline(item, loadingInfo))
+  if (!isTypeLoadProtected(loadingInfo) && loadingInfo.onlyChildren) {
```

Comment:
I strongly prefer to move this logic to the component. So just load all disciplines here, no need to pass "onlyChildren" or anything.

I recommend:
Create a new hook useLoadAllDisciplines
Call this hook from the component and filter the root disciplines in the component.